api/new-contribution.ts

```diff
@@ -0,0 +1,143 @@
+import {VercelRequest, VercelResponse} from '@vercel/node';
+import axios from 'axios';
+import OpenAI from 'openai';
+import toMarkdown from '@sanity/block-content-to-markdown';
+import {writeClient, readClient} from './curate-contribution';
+
+const slackWebhookUrl = process.env.SPAM_RATING_SLACK_WEBHOOK_URL;
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY,
+});
+
+const TOKEN_LIMIT = process.env.OPENAI_TOKEN_LIMIT || 4097;
+const SUBMIT_TO_OPENAI = true;
+const WRITE_TO_SANITY = false;
+const POST_TO_SLACK = false;
+
+const stopwords = new Set([
+  'a',
+  'an',
+  'the',
+  'and',
+  'or',
+  'but',
+  'is',
+  'if',
+  'then',
+  'else',
+  'when',
+]);
+
+function filterText(text: string) {
+  return text
+    .split(/\s+/)
+    .filter((word) => {
+      if (word.startsWith('http') || word.includes('http') || !stopwords.has(word.toLowerCase())) {
+        return true;
+      }
+      return false;
+    })
+    .join(' ');
+}
+
+function chunkText(text: string, tokenLimit: number): string[] {
+  const words = text.split(' ');
+  let chunk = '';
+  let chunks: string[] = [];
+
+  for (let word of words) {
+    if ((chunk + word).length > tokenLimit) {
+      chunks.push(chunk.trim());
+      chunk = '';
+    }
+    chunk += ' ' + word;
+  }
+
+  if (chunk.length > 0) {
+    chunks.push(chunk.trim());
+  }
+
+  return chunks;
+}
+
+async function getSpamScore(title: string, body: string, threshold: number, tokenLimit: number) {
+  const paragraphs = body.split('\n\n'); // Splitting by paragraphs
+  for (const paragraph of paragraphs) {
+    const filteredParagraph = filterText(paragraph);
+    const subParagraphs = chunkText(filteredParagraph, tokenLimit - title.length);
+
+    for (const subParagraph of subParagraphs) {
+      try {
+        if (!SUBMIT_TO_OPENAI) {
+          console.log({role: 'user', content: `title: ${title}\npart of body: ${subParagraph}`});
+        }
+        
+        if (SUBMIT_TO_OPENAI) {
+          const chatCompletion = await openai.chat.completions.create({
+            messages: [
+              {
+                role: 'system',
+                content: `You are a helpful project gallery forum moderator. I will give you a title, and part of the body of a post, and you will return a rating of 1-7 of how likely it is to be spam, along with the reasons for why you either consider it safe or a high chance of it being spam as structured JSON format like {"rating": <number>, "reasons": ["reason1", "reason2"]}`,
```

Comment:
I guess we don't need to worry that much about prompt injection. I mean, if someone goes to that length to post spam, we'll deal with then?